fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/RecordMetaDataBuilderTest.java

```diff
@@ -224,4 +227,58 @@ public void badUnionFields() {
                 RecordMetaData.newBuilder().setRecords(TestRecordsBadUnion2Proto.getDescriptor()));
         assertEquals("Union field _MySimpleRecord should not be repeated", t.getMessage());
     }
+
+    @Test
+    public void localMetaData() {
+        // The original meta-data
+        RecordMetaDataProto.MetaData originalMetaData = RecordMetaData.build(TestRecords1Proto.getDescriptor()).toProto(null);
+
+        // Make a local version of the meta-data dynamically
+        RecordMetaDataProto.MetaData.Builder serializedLocalMetaDataBuilder = RecordMetaDataProto.MetaData.newBuilder()
+                .setRecords(originalMetaData.getRecords());
+        serializedLocalMetaDataBuilder.addDependencies(TestRecords2Proto.getDescriptor().toProto());
+        serializedLocalMetaDataBuilder.addRecordTypes(serializedLocalMetaDataBuilder
+                                                            .addRecordTypesBuilder()
+                                                            .setName("MyLongRecord").build());
+        serializedLocalMetaDataBuilder.setRecords(serializedLocalMetaDataBuilder
+                                                            .getRecordsBuilder()
+                                                            .addMessageType(TestRecords2Proto.MyLongRecord.getDescriptor().toProto())
+                                                            .build());
```

Comment:
The `MyLongRecord` record type is not being added to the meta-data's union descriptor, which means that the new meta-data is malformed. That's a fine thing to test, but we should also test that it actually works when everything is updated correctly.

Reply:
How do you add it to the union dynamically?

Reply:
added this statically.